### PR TITLE
feat(components/form/builder): add applyEmptyFields prop

### PR DESCRIPTION
**Pull request by javiauso**
add applyEmptyFields prop

### CODE REVIEW

**Review by Sabri1209 — APPROVED** [review]

**Review by andresz1 — APPROVED** [review]

File: components/form/builder/src/index.js [hunk 1]
```diff
@@ -178,20 +183,6 @@ FormBuilder.propTypes = {
   children: PropTypes.func
 }
 
-FormBuilder.defaultProps = {
```
Comment (jordevo) [inline components/form/builder/src/index.js:181]: 👏

File: components/form/builder/src/index.js [hunk 2]
```diff
@@ -18,21 +18,22 @@ import AtomSpinner, {AtomSpinnerTypes} from '@s-ui/react-atom-spinner'
 import {getUpdatedFormState} from './mapper/formState'
 
 const FormBuilder = ({
+  applyEmptyFields,
```
Comment (jordevo) [inline components/form/builder/src/index.js:21]: what is the new prop about? by its name, I'm not sure it's self-explanatory. Also, should it default to `false`?

**Review by tomasmax — APPROVED** [review]

**Review by jordevo — APPROVED** [review]

File: components/form/builder/src/index.js (hunk 2 above)
Comment (javiauso) [reply components/form/builder/src/index.js:21]: This prop apply form builder logic like apply rules and set initialFields values for empty fields, for example, if has defined `{categoryId: '1', subCategoryId: ''}` without `applyEmptyFields` prop, will only apply rules and form logic for categoryId. The name probably could be improved, any suggestion 😄?

File: components/form/builder/src/index.js [hunk 3]
```diff
@@ -18,21 +18,22 @@ import AtomSpinner, {AtomSpinnerTypes} from '@s-ui/react-atom-spinner'
...
   json,
-  initFields,
-  onChange,
-  onFocus,
-  onBlur,
-  responseInterceptor,
-  requestInterceptor,
+  initFields = {},
+  onChange = () => {},
```
Comment (andresin87) [inline components/form/builder/src/index.js:24]: Functions should not be empty (Code Smell –Critical)
https://rules.sonarsource.com/javascript/RSPEC-1186

File: components/form/builder/src/index.js (hunk 2 above)
Comment (javiauso) [reply components/form/builder/src/index.js:21]: By default is falsy to not break actual behaviour

Comment (jordevo) [reply components/form/builder/src/index.js:21]: my 2 cents if I may 😉

`shouldAlwaysApplyRules`
`forceRulesOnAllFields`

I miss the `rules` part on `applyEmptyFields` and I think it's key to understand the purpose of the prop...

**Review by kikoruiz — APPROVED** [review]

File: components/form/builder/src/index.js (hunk 2 above)
Comment (javiauso) [reply components/form/builder/src/index.js:21]: I like `forceRulesOnAllFields`, I think it's clearer, thank you!

**Pull request merged**